src/sensor/src/Sensor.cpp

```diff
@@ -65,23 +71,23 @@ bool Sensor::receive_cc_config_msg(uint8_t *buf, size_t bufSize) {
     for(auto& block : reader.blocks) {
         if(block->type == BlockType::cnt_sensor_config) {
             auto configBlock = reinterpret_cast<CntSensorConfigBlock*>(block.get());
-
-            reload_config(configBlock->port_id);
-
-            central_ips.clear();
-            for(auto& cnt_ip : configBlock->central_ips) {
-                central_ips.push_back(cnt_ip);
-            }
-
+            mutex.lock();
+            reload_config(configBlock->cnt_ip, configBlock->port_id, configBlock->central_ips);
+            mutex.unlock();
             log() << "Message: " << configBlock->toString();
         }
     }
     return true;
 }
 
+void Sensor::reload_config(std::string& cc_ip, in_port_t cc_port, std::vector<std::string>& central_ips) {
+    config.cc_addr = cc_ip;
```

Comment:
Ta metoda powinna zakładać locka na mutexa. Dzięki temu nie musisz tego powtarzać w każdym potencjalnym miejscu wywołania i nie robisz wymagań wobec kontekstu (których nie widać bez spojrzenia w kod). Dodatkowo wydaje się to bardziej naturalne (gwarancja atomowości operacji).